src/restconfig/src/main/java/org/geoserver/rest/catalog/WorkspaceController.java

```diff
@@ -175,7 +175,7 @@ public void workspacePut(
             }
 
             String infoName = workspace.getName();
-            if (infoName != null && !workspaceName.equals(infoName)) {
+            if (infoName != null && infoName.isEmpty()) {
                 throw new RestException("Can't change name of workspace", HttpStatus.FORBIDDEN);
```

Comment:
The error message should also be updated to something like "the name cannot be empty" or something like that. 
Honestly I don't know why the check was here, as the UI allows for workspace renames.

Reply:
Checked the history, looks like this behavior was there since the very beginning of the REST API, 9 years ago:
https://github.com/geoserver/geoserver-history/blame/master/src/restconfig/src/main/java/org/geoserver/catalog/rest/WorkspaceResource.java